### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by sv-Tjipto**
A Siamese Networking model to classify Alzheimer's disease using ADNI brain data.

Using the train.py to run the program

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- sufficient information on algorithm and usage
- no information on data splits, or augmentations
- a loss plot is shown (assuming it's for the training data)
- no discussion on the results

Commit messages: OK

Code: 
- good design
- sufficiently commented

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification**
- seem to achieve OK accuracy for similarity detection with the siamese

General comments:

- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem, no single image classifier -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, splits unknown -1
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe acceptable, could have more info like discussion -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove pyc files before I can merge -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
